Helper/VarHelper.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Sailthru\MageSail\Helper;
+
+class VarHelper
+{
+    private $vars = [
+        'firstname' => ['first', 'name'],
+        'middlename' => ['middle', 'name'],
+        'lastname' => ['last', 'name'],
+    ];
+    private function formatCamelCaseVars()
+    {
+        $ret = [];
+        foreach($this->vars as $k => $v)
+        {
+            $str = $v[0];
+            for ($i = 0; $i < count($v) - 1; $i++)
+            {
+                $str = $str . ucfirst($v[$i + 1]);
+            }
+            $ret[$k] = $str;
+        }
+        return $ret;
+    }
+    private function formatSnakeCaseVars()
+    {
+        $ret = [];
+        foreach($this->vars as $k => $v)
+        {
+            $ret[$k] = implode("_", $v);
+        }
+        return $ret;
+    }
```

Comment:
I think there's still too much logic within each of these functions. It'd be a bit cleaner if `formatCamelCase` and `formatSnakeCase` could be dumb one-liners, leaving the iteration logic to the public function.

Reply:
This is hard to do with camelCase the way it's designed, though. If the goal is quick and and clean, the `1 ... count($array)` indices can be capitalized, i.e.
```
['first', 'Name']
['last', 'Name']
```
etc,
which makes this a lot easier.

Reply:
Sorry, was more referring to each of the keys. e.g. only _one_ function should have logic to iterate over all the keys and make the transformations. The transformation functions should only take relevant args for a single transformation. E.g.

`formatSnakeCaseVar( [ "snake", "case" ] );`

with the main method taking care of the iteration.

And to reduce logic for the camel case function you can use `array_shift` to pop the first element and concatenate to the mapped/joined rest. You can use an array popping method to grab the first element and combine with the mapped rest as well. https://www.php.net/manual/en/function.array-shift.php